src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -61,7 +61,7 @@ case class DepositIngestTask(deposit: Deposit, dansBagValidator: DansBagValidato
           """.stripMargin)
       }
       ddm <- deposit.tryDdm
```

Comment:
You are passing something extracted from teh deposit (ddm) and the whole deposit. Not really intuitive! You are no probably tempted to solve this by extracting the ddm inside toDataverseDataset but please don't! The other way around is much better and allows you to write better unit tests. So: extract the vault metadata values from deposit.properties in the line above this one (into a case class for example) and then pass that as the second argument. You will see that your unit test now also become much more readable, as you can create and pass the VaultMetadata case class object in the unit test instead of referencing a deposit somewhere in a different directory.

Reply:
I agree...

Reply:
OK, this looks better. In the unit test you are still reading the vault metadata via the Deposit. One of the benefits that I mentioned about is that that is not longer necessary. You can just create `VaultMetadata` object in the unit test itself. The advantage of that is that you can see in the unit test itself what the expected values are. Now you have to refer to a file somewhere that you first have to look up. I'll merge this now, as I want to move on.

Reply:
I had it first the way you describe.  The reason that I changed it back to reading via a deposit is that in that way you also test that fetching the vault metadata goes correctly.

Reply:
It is a common mistake to want to "touch" as many things as possible in a unit test. I think that you should have only one such a test (a "smoke test", if I am not mixing up terminology here) and that the rest should be small and focused, so that if unit test N fails you know exactly where the problem is right away. If the big test, if it fails, will only tell you that "something" went wrong.

Reply:
I was already expecting this answer:)  And you are correct, I also already thought that it would be better to have a separate test to check that fetching the metadata goes correctly.